asserts/asserts.go

```diff
@@ -65,6 +83,7 @@ type Assertion interface {
 
 // assertionBase is the concrete base to hold representation data for actual assertions.
 type assertionBase struct {
+	// TODO: worth having a type *AssertionType cache field now?
```

Comment:
Probably not for now, given how seldom used it is. Would be more trouble than it's worth.